internal/rbd/rbd_util.go

```diff
@@ -532,7 +532,13 @@ func (ri *rbdImage) isInUse() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	ri.Primary = mirrorInfo.Primary
+
+	if mirrorInfo.State == librbd.MirrorImageEnabled {
+		if !ri.Primary {
+			return false, fmt.Errorf("image %s is not primary", ri)
```

Comment:
Should this not return `true`?

It would be good to update the comment of the function too. It explains only about watchers, and a note about mirroring (requiring it to be the primary) would be good.

Reply:
This function returns false for all the error cases.

Reply:
Not being primary is not an error. It is a valid state. Just like there could be more watchers than expected. The image is just "in use", and should not be mapped on the node.

Reply:
If you prefer to report details about why the image is "in use", consider cleaning up this function by only returning an error, and not an additional bool.

Reply:
>If you prefer to report details about why the image is "in use", consider cleaning up this function by only returning an error, and not an additional bool.

This requires more changes as we try to map with exponential backoff. altered function to log error if the image is not primary and return true,nil to try in exponential backoff.

Reply:
ah, ok! in that case, document the behaviour in the description of the function too :-)

Reply:
Now I remember why I made this change, The bool value (watchers) are/is discarded if the request is for RWX volume. Mapping secondary can cause issues that's why we are introducing this check, reverting this change PTAL